pig/src/main/java/org/jboss/pnc/bacon/pig/impl/addons/scanservice/PostBuildScanService.java

```diff
@@ -0,0 +1,196 @@
+                validateMapKeys(subMap);
+            }
+        }
+    }
+
+    void checkPresenceOfMandatoryFields(Map<String, ?> map) throws ConfigMissingException {
+        for (String item : mandatoryItems) {
+            List<String> items = map.keySet()
+                    .stream()
+                    .filter(k -> k.trim().contains(item))
+                    .collect(Collectors.toList());
+            if (items.size() != 1) {
+                Validate.fail(" mandatory field '" + item + "' not provided");
+            }
+        }
+    }
+
+    /* Validate our addon config */
+    @Override
+    public void validate() {
+        try {
+            validateMapKeys(postBuildScanConfigMap);
+            checkPresenceOfMandatoryFields(postBuildScanConfigMap);
+            Validate.validateUrl((String) postBuildScanConfigMap.get("serviceUrl"), "PSSaaS");
+        } catch (ConfigMissingException e) {
+            // We can't handle incorrect configuration, terminate the app with a runtime exception
+            throw new RuntimeException(e);
+        }
+    }
+
+    private ScanServiceDTO createScanServiceDTO() {
+        String s = postBuildScanConfigMap.get("productID").toString();
+        String productId = s;
+        if (s == null || s.isEmpty()) {
+            log.error("Missing mandatory parameter Product ID");
+            return null;
+        }
+        s = (String) postBuildScanConfigMap.get("EventId");
+        String eventId = (s != null && !s.isEmpty()) ? s : ScanServiceDTO.EVENT_ID_DFT_VALUE;
+
+        s = (String) postBuildScanConfigMap.get("cpaasVersion");
+        String cpaasVersion = (s != null && !s.isEmpty()) ? s : ScanServiceDTO.CPAAS_VERSION_DFT_VALUE;
+
+        s = (String) postBuildScanConfigMap.get("jobUrl");
+        String jobUrl = (s != null && !s.isEmpty()) ? s : ScanServiceDTO.JOB_URL_DFT_VALUE;
+
+        Boolean b = (Boolean) postBuildScanConfigMap.get("isManagedService");
+        Boolean isManagedService = (b != null) ? b : ScanServiceDTO.IS_MANAGED_SERVICE_DFT_VALUE;
+
+        List<Integer> brewBuilds = (List<Integer>) postBuildScanConfigMap.get("brewBuilds");
+
+        List<Map<String, String>> extraSCMURLs = (List<Map<String, String>>) postBuildScanConfigMap.get("extraScmUrls");
+
+        return new ScanServiceDTO(
+                productId,
+                eventId,
+                isManagedService,
+                cpaasVersion,
+                jobUrl,
+                brewBuilds,
+                extraSCMURLs,
+                builds);
+    }
+
+    private String showDTOfields(ScanServiceDTO ssDTO) throws JsonProcessingException {
+        return new ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(ssDTO);
+    }
+
+    private void sendScanRequest(ScanServiceDTO ssDTO) {
+        String dto = null;
+        try {
+            dto = showDTOfields(ssDTO);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```

Comment:
Surely this is an error if it can't be parsed?

Reply:
If ObjectMapper().writerWithDefaultPrettyPrinter().writeValueAsString(ssDTO) cannot parse a given object it looks like a problem with that object.

Initially a method sendScanRequest was conceived as an entry point to handle client request. Later that code was put directly in the trigger method. As a result sendScanRequest is more like a placeholder at the moment. To do something useful it currently dumps the DTO contents. One day we might move that code from trigger to it.

Reply:
I'm not sure I follow - perhaps we discuss on gchat.

Reply:
Fixed in https://github.com/project-ncl/bacon/pull/946/commits/a03f209a9e9c0a51903953e1f9c0264506cf3cb6 by no longer printing stack trace